lib/ukargparse/Config.uk

```diff
@@ -2,30 +2,13 @@ menuconfig LIBUKARGPARSE
 	bool "ukargparse: Simple argument parser"
 	default n
 	select LIBUKDEBUG
+	select LIBUKARGPARSE_TEST if LIBUKTEST_ALL
 
 if LIBUKARGPARSE
 
-config LIBUKARGPARSE_TESTS
+config LIBUKARGPARSE_TEST
 	bool "Enable tests"
 	default n
-	select LIBUKTEST
-	select LIBUKARGPARSE_TESTS_PARSE if LIBUKTEST_ALL
-	select LIBUKARGPARSE_TESTS_NULL if LIBUKTEST_ALL
-	select LIBUKARGPARSE_TESTS_EXTRA if LIBUKTEST_ALL
+	depends on LIBUKTEST
```

Comment:
@nderjung, @marcrittinghaus, should a unit test option of a library `depend on` `LIBUKTEST` or should it `select` `LIBUKTEST`? I am actually thinking it should be the latter. What do you think?

Reply:
Hi, I think it would be best if it would select `UKTEST`, not depend on it, but it creates a circular dependency error since I want to select the test suite if `UKTEST_ALL` is selected. I didn't manage to find a workaround exept change it to `depends on`, if you have any ideas let me know.

Reply:
Been blocking on a similar PR on libsodium [due to the same question](https://github.com/unikraft/lib-libsodium/pull/3#issuecomment-1046219236). Perhaps it would be useful to have a universal policy for all libraries?

Btw, `depends on` has also the drawback that the option to enable testing in a given library magically appears if you have enabled `CONFIG_UKTEST`, otherwise it is not evident to a user not familiar with UK internals that the library provides tests at all.

Reply:
Actually, `UKTEST_ALL` should influence the header files and macros. Actually, similar to `uk_pr_debug`, I assumed that enabling/disabling tests is done behind the scenes by defining a compile flag or done by `lib/uktest` headers automatically. Is this the case @nderjung , @marcrittinghaus ?

Reply:
That makes sense, but I think it would be a bit weird to have the test suite active because of the `UKTEST_ALL` option but look disabled from the menuconfig interface.

Reply:
No, this is fine. The same happens with the debug option. Actually, only a few libs have it implemented in the menu currently (and the plan was always to add a `Enable debug messages` option to each lib) but the global option should overwrite individual settings. I think if we keep consistency across the different debugging options (debug print, tracepoints, unit tests, ...), it is not confusing. Instead, I am convinced that we cause confusing if our systems work all differently.

Reply:
@skuenzer so should I just change the `#if CONFIG_LIBUKARGPARSE_TEST` to `#if CONFIG_LIBUKARGPARSE_TEST || CONFIG_LIBUKTEST_ALL`? I looks like it should do the trick, but for whatever reason it does not work, if I enable the `LIBUKTEST_ALL` option it just runs the self tests on `uktest`.

Reply:
Actually, I don't know what @nderjung's original idea was on how `CONFIG_LIBUKTEST_ALL` should work. But here are my thoughts on this:

* I would definitely go with `select UKLIBTEST` for manual test activation in this library. I agree with @michpappas that it is totally confusing to have options magically appear.
* I would prefer to also have the menu items selected when `LIBUKTEST_ALL` is active just like it is done in the PR. I like that solution and it is a pity that it does not work. `kconfig` is really limited in some ways 😭 
* The guard does not work as expected because the `Makefile` includes the respective source file only if `CONFIG_LIBUKARGPARSE_TEST` is selected.

So we have two options here:
1) Remove the guard and do the check in the Makefile like so:
```
ifneq ($(filter y,$(CONFIG_LIBUKARGPARSE_TEST) $(CONFIG_LIBUKTEST_ALL)),)
LIBUKARGPARSE_SRCS-y += $(LIBUKARGPARSE_BASE)/tests/test_ukargparse_parse.c
endif
```
2) Keep the guard and always include the source file.

I would probably go with solution 1 to reduce build time.

Reply:
:thinking: I was expecting that tests are automatically compiled-out by uktest with the macros/functions that we have and without needing any guards in the sources or dependencies in `Config.uk`. Similar to `UK_DEBUG` the compile flag (provided through `Makefile.uk`) that you could set per file you could enable the unit tests with a library config option. The other alternative is that within `lib/ukutest` you can enable all tests. But from the file defining the tests, it should be possible to always include it and not needing to do any manual instrumentation in the file. Is this the case @nderjung?
In general, if this is the case, we actually always need to select `uktest` so that the hiding macros work and/or all test can be enabled with one config option in uktest...

Reply:
Yes, `uktest` is not well integrated in this regard, yet However, you also need to have `.c` files from `uktest` built so you would have to always select `uktest` if you do not want to make explicit `select LIBUKTEST` statements in `Config.uk`. 

But I don't really like that solution. So I would still keep the explicit `select LIBUKTEST` for config options in the library to enable tests.

Say, `UK_TESTCASE` etc. were conditionally defined empty if `UK_TEST` and `CONFIG_LIBUKTEST_ALL` were not specified (just like with the tracepoints). Then you always need a:
```C
#if CONFIG_LIBARGPARSE_TEST
#define UK_TEST
#endif
#include <uk/test.h>
```
Or something like this in the Makefile:
```
ifeq ($(CONFIG_LIBARGPARSE_TEST),y)
LIBUKARGPARSE_CFLAGS-y     += -DUK_TEST
LIBUKARGPARSE_CXXFLAGS-y += -DUK_TEST
endif
```

Reply:
A disadvantage of this simple solution is that you would always have the source files of the tests built (although mostly empty then).

Reply:
:thinking: Hum. okay, I understand. Yes, it is true that we would always include kind of empty source files if `ukutest` is disabled but we could still use a filter statement in the `Makefile.uk` of the library as you suggested before. This could be something to do in the cases where it would otherwise become too ugly. However, this would be an optional optimization. Additonally, it would also cleanly enable us the case where tests and implementation are placed in the same source file (independent of if this is good or bad practice, it could be possible that we enter some day in a situation where this ability is required).
My suggestion on how to proceed with this PR now: I think we should make the changes as @marcrittinghaus  suggested with the comment earlier (the one with filtering in `Makefile.uk`). Adopting `uktest` should be done with another PR to get it inline more inline with the [debug printing enable/disable approach](https://youtu.be/IFoLHi02-Iw?t=347). We then can come back to update this (in the meantime accepted) commit. What do you guys think?

Reply:
I think that is the best thing to do now. If you are all ok with this let me know and I will push the changes suggested by @marcrittinghaus so you can review it.

Reply:
@StefanJum Yes, just go ahead so that we can get it into the release today